ViewModel/FrequencyRelationships/RetrieveGraphicalStageFrequencyVM.cs

```diff
@@ -0,0 +1,126 @@
+        #endregion
+        #region Constructors
+        public RetrieveGraphicalStageFrequencyVM()
+        {
+            Initialize();
+            GenerateFrequencyCurves = new NamedAction();
+            GenerateFrequencyCurves.Name = "GenerateFrequencyCurves";
+            GenerateFrequencyCurves.Action = GenerateFrequencyCurvesAction;
+        }
+        #endregion
+        #region Methods
+        private void Initialize()
+        {
+            AvailableHydraulics = new ObservableCollection<HydraulicElement>();
+            List<HydraulicElement> hydraulicElements = StudyCache.GetChildElementsOfType<HydraulicElement>();
+            foreach(HydraulicElement hydraulic in hydraulicElements)
+            {
+                AvailableHydraulics.Add(hydraulic);
+            }
+            SelectedHydraulics = AvailableHydraulics[0];
+
+            AvailableIndexPointSets = new ObservableCollection<IndexPointsElement>();
+            List<IndexPointsElement> indexptsElements = StudyCache.GetChildElementsOfType<IndexPointsElement>();
+            foreach(IndexPointsElement indexpt in indexptsElements)
+            {
+                AvailableIndexPointSets.Add(indexpt);
+            }
+            SelectedIndexPointSet = AvailableIndexPointSets[0];
+        }
+
+        private void GenerateFrequencyCurvesAction(object arg1, EventArgs arg2)
+        {
+
+            string pointShapefile = Storage.Connection.Instance.IndexPointsDirectory + "\\" + SelectedIndexPointSet.Name + "\\" + SelectedIndexPointSet.Name + ".shp";
+            string hydraulicParentDirectory = Storage.Connection.Instance.HydraulicsDirectory+"\\"+SelectedHydraulics.Name;
+            List<UncertainPairedData> freqCurves = SelectedHydraulics.DataSet.GetGraphicalStageFrequency(pointShapefile,hydraulicParentDirectory );
+            for(int i =0; i < freqCurves.Count; i++)
+            {
+                AddFrequencyRelationship(freqCurves[i], SelectedIndexPointSet.IndexPoints[i]+ "|" + SelectedHydraulics.Name);
+            }
+        }
+        private void AddFrequencyRelationship(UncertainPairedData upd, string name)
+        {
+            string editDate = DateTime.Now.ToString("G"); //will be formatted like: 2/27/2009 12:12:22 PM
+            int id = PersistenceFactory.GetElementManager<AnalyticalFrequencyElement>().GetNextAvailableId();
+            //Create graphical VM
+            GraphicalVM graphicalVM = new GraphicalVM(StringConstants.GRAPHICAL_FREQUENCY, StringConstants.EXCEEDANCE_PROBABILITY, StringConstants.DISCHARGE);
+            graphicalVM.Options[0].UpdateFromUncertainPairedData(upd);
+
+            AnalyticalFrequencyElement element = new AnalyticalFrequencyElement(name, editDate,"Retrieved from Hydraulics",DefaultData.PeriodOfRecord, false, true, DefaultData.LP3Mean,DefaultData.LP3StDev, DefaultData.LP3Skew,new List<double>(), graphicalVM,new CurveComponentVM(), id);
```

Comment:
AnalyticalFrequencyElement is curiously named. AnalyticalFrequencyElement applies to both analytical functions and graphical functions?

Reply:
It does. frustrating for sure. It's a relic of the old FDA before we got our hands in it. and before there was a graphical. I'd love to do a huge rework of this ui element, but it's just not worth priority right now.